src/commit.c

```diff
@@ -19,6 +19,7 @@
 #include "message.h"
```

Comment:
I think the commits are out of order. You first use the new generic mechanism and introduce it only in the next commit

Reply:
Yes, I caused some historic wreckage when doing some rebases, so when date-sorting, the commits are reverted. I already tried to fix it but failed, so I think I'll end up extracting patches and reapplying manually.